### PR TITLE
add extra autoscaler flag support

### PR DESCRIPTION
**Pull request by marwanad**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by marwanad** [comment]
@markcowl can you help merge this into the dev branch?

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json [hunk 1]
```diff
@@ -1980,6 +1980,9 @@
         "autoScalerProfile": {
           "type": "object",
           "properties": {
+            "balance-similar-node-groups": {
```
Comment (markcowl) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json]: I believe this is a breaking change requiring a new api-version by ARM standards.  Looping in ARM folks to verify

Comment (marwanad) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json]: yup, we're consolidating all of our changes against a dev branch then we'll target master with the new api version.

**Comment by marwanad** [comment]
@markcowl can you ping someone from ARM?

**Review by xizha162 — APPROVED** [review]

**Comment by xizhamsft** [comment]
@ravbhatnagar, could you take a look of this simple change?

**Review by markcowl — APPROVED** [review]

**Comment by ravbhatnagar** [comment]
Looks like this change will be made in a new api-version when moving to master, which seems reasonable to me.

**Pull request merged**